pkg/apis/batch/validation/validation.go

```diff
@@ -206,6 +208,9 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 			allErrs = append(allErrs, field.Required(fldPath.Child("backoffLimitPerIndex"), fmt.Sprintf("when maxFailedIndexes is specified")))
 		}
 	}
+	if spec.ManagedBy != nil {
+		allErrs = append(allErrs, apivalidation.ValidateDNS1123Subdomain(*spec.ManagedBy, fldPath.Child("managedBy"))...)
```

Comment:
Discussion with liggitt and michal via slack.  Validation to be updated to `IsDomainPrefixedPath` (matches IngressClass

Reply:
Done, used this for validation now, I also check the length of the field is max 63. Otherwise it could be an arbitrarily long string, taking 2MB is some pathological cases.